sourcegraph/sourcegraph.proto

```diff
@@ -588,6 +588,215 @@ service Repos {
 	rpc ListCommitters(ReposListCommittersOp) returns (CommitterList);
 }
 
+// StorageError represents an error when interacting with the Storage service.
+message StorageError {
+	// Code represents the type of error for programatic handling.
+	enum Code {
+		// NONE is the error code used when programatic handling of the error is not
+		// advised. It can be assumed that the operation failed, and that a human
+		// readable message is also provided.
+		NONE = 0;
+
+		// EOF is the error returned by Read when no more input is available.
+		// Functions should return EOF only to signal a graceful end of input. If
+		// the EOF occurs unexpectedly in a structured data stream, the appropriate
+		// error is either UNEXPECTED_EOF or some other error message giving more
+		// detail.
+		EOF = 1;
+
+		// NOT_EXIST is the error used when attempting to read/write to an object
+		// that does not exist.
+		NOT_EXIST = 2;
+
+		// PERMISSION is the error used when permission to access the given file is
+		// not granted to you.
+		PERMISSION = 3;
```

Comment:
The naming is not consistent with the rest of the repository (e.g., see [here](https://github.com/sourcegraph/go-sourcegraph/blob/a246b02e2e9e353772d25f7f24230b08c5eb87df/sourcegraph/sourcegraph.proto#L3065-L3078)). It should follow Go style, i.e., `NotExist`, `Permission`.